src/librustc_trans/debuginfo/gdb.rs

```diff
@@ -22,19 +23,13 @@ use std::ptr;
 use syntax::attr;
 
 
-/// Inserts a side-effect free instruction sequence that makes sure that the
-/// .debug_gdb_scripts global is referenced, so it isn't removed by the linker.
-pub fn insert_reference_to_gdb_debug_scripts_section_global(ccx: &CrateContext, builder: &Builder) {
+/// Inserts the .debug_gdb_scripts global into the set of symbols
+/// to be placed in `llvm.used`, so it isn't removed by the linker.
```

Comment:
To add to this, the default linker script on x86_64 Linux has no mention of `.debug_gdb_scripts` (or a wildcard over all `.debug*` sections), and the logic of --gc-sections does not appear to special-case any of that either (it only really special-cases `.eh_frame`, in general; the rest is handled by traversing the graph). So this is interesting.